### PR TITLE
When a subquery is included in an aggregate function, the subquery is parsed multiple times

### LINKED ISSUE #5094: When a subquery is included in an aggregate function, the subquery is parsed multiple times
### Steps to reproduce the behavior (Required)
select sum(case when v4 = (select v1 from t0) then v4 end) from t1
### Expected behavior (Required)
build apply node only once
### Real behavior (Required)
build apply node twice

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

### PR DESCRIPTION
**Pull request by HangyuanLiu**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #5094

### CODE REVIEW

**Comment by HangyuanLiu** [comment]
Aggregate functions are resolved at the stage of processing aggregate nodes. At this time, if the aggregate function contains a subquery, the subquery will be converted to an ApplyNode. However, the aggregate function also belongs to the outputExpression, then at this time, the subquery in the aggregate function will generate an ApplyNode again. Caused ApplyNode to appear twice, affecting the execution of xiao'l.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 4 / 4 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/transformer/SubqueryTransformer.java|4|4|100.00%|

File: fe/fe-core/src/test/java/com/starrocks/sql/plan/AggregateTest.java [hunk 1]
```diff
@@ -691,4 +691,27 @@ public void testLocalAggregateWithMultiStage() throws Exception {
...
         FeConstants.runningUnitTest = false;
     }
+
+    @Test
+    public void testAggWithSubquery() throws Exception {
+        String sql = "select sum(case when v4 = (select v1 from t0) then v4 end) from t1";
+        String plan = getFragmentPlan(sql);
+        Assert.assertTrue(plan.contains("  7:AGGREGATE (update finalize)\n" +
+                "  |  output: sum(8: case)\n" +
+                "  |  group by: \n" +
```
Comment (Seaven) [inline fe/fe-core/src/test/java/com/starrocks/sql/plan/AggregateTest.java:701]: The assert don't check the result which you want

File: fe/fe-core/src/test/java/com/starrocks/sql/plan/AggregateTest.java [hunk 2]
```diff
@@ -691,4 +691,27 @@ public void testLocalAggregateWithMultiStage() throws Exception {
...
+                "  |  cross join:\n" +
+                "  |  predicates is NULL.\n" +
+                "  |  \n" +
+                "  |----4:EXCHANGE\n" +
+                "  |    \n" +
+                "  2:ASSERT NUMBER OF ROWS\n" +
+                "  |  assert number of rows: LE 1\n" +
+                "  |  \n" +
+                "  1:EXCHANGE"));
+    }
```
Comment (Seaven) [inline fe/fe-core/src/test/java/com/starrocks/sql/plan/AggregateTest.java:716]: I think the assert don't check the result which you want

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 4 / 4 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/transformer/SubqueryTransformer.java|4|4|100.00%|

**Review by kangkaisen — DISMISSED** [review]

**Review by Seaven — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 4 / 4 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/transformer/SubqueryTransformer.java|4|4|100.00%|

**Review by liuyehcf — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**